src/helpers.rs

```diff
@@ -846,6 +846,17 @@ pub trait EvalContextExt<'tcx>: crate::MiriInterpCxExt<'tcx> {
         self.set_last_error(self.io_error_to_errnum(err)?)
     }
 
+    /// Sets the last OS error using a `std::io::ErrorKind` and writes -1 to dest place.
+    fn set_io_error_and_return_neg_one(
+        &mut self,
+        err: io::Error,
```

Comment:
If you use `err: impl Into<io::Error>`, some of the call sites should become a bit simpler.

Reply:
Wow, yea this is better, now I can do ``ecx.set_last_error_and_return(ErrorKind::BrokenPipe, dest)``

Reply:
Exactly. :)